sklearnex/covariance/incremental_covariance.py

```diff
@@ -14,18 +14,30 @@
 # limitations under the License.
 # ===============================================================================
 
+import numbers
+
 import numpy as np
+from scipy import linalg
+from sklearn.covariance import EmpiricalCovariance as sklearn_EmpericalCovariance
 from sklearn.utils import check_array, gen_batches
 
 from daal4py.sklearn._n_jobs_support import control_n_jobs
+from daal4py.sklearn._utils import sklearn_check_version
 from onedal._device_offload import support_usm_ndarray
 from onedal.covariance import (
     IncrementalEmpiricalCovariance as onedal_IncrementalEmpiricalCovariance,
 )
+from sklearnex._device_offload import dispatch, wrap_output_data
+from sklearnex._utils import PatchingConditionsChain, register_hyperparameters
+from sklearnex.metrics import pairwise_distances
+
+if sklearn_check_version("1.2"):
+    from sklearn.utils._param_validation import Interval
 
 
-@control_n_jobs(decorated_methods=["partial_fit"])
-class IncrementalEmpiricalCovariance:
+# TODO: consult with others whether this should support store_precision and assume_centered
+@control_n_jobs(decorated_methods=["partial_fit", "fit"])
```

Comment:
decorated_methods should be ["partial_fit", "_onedal_finalize_fit"]. there is no need to decorate fit since it does not call onedal directly, and _onedal_finalize_fit does